src/main/java/duke/MainWindow.java

```diff
@@ -0,0 +1,65 @@
+package duke;
+
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+
+
+/**
+ * Controller for MainWindow. Provides the layout for the other controls.
+ */
+public class MainWindow extends AnchorPane {
+    @FXML
+    private ScrollPane scrollPane;
+    @FXML
+    private VBox dialogContainer;
+    @FXML
+    private TextField userInput;
+    @FXML
+    private Button sendButton;
+
+    private Duke duke;
+
+    private Image userImage = new Image(this.getClass().getResourceAsStream("/images/user.png"));
+    private Image dukeImage = new Image(this.getClass().getResourceAsStream("/images/duke.png"));
+
+    @FXML
+    public void initialize() {
+        scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
+    }
+
+    public void setDuke(Duke d) {
+        duke = d;
+        dialogContainer.getChildren().add(DialogBox.getDukeDialog(Ui.showWelcome(), dukeImage));
+    }
+
+    /**
+     * Creates two dialog boxes, one echoing user input and the other containing Duke's reply and then appends them to
+     * the dialog container. Clears the user input after processing.
+     */
+    @FXML
+    private void handleUserInput() {
+        String response = "";
+        String input = "";
+        String color = "#6adcd3";
+        try {
+            input = userInput.getText();
+            response = duke.getResponse(input);
+        } catch (DukeException e) {
+            color = "#ffb6c1";
```

Comment:
```suggestion
        String final COLOR_BLUE = "#6adcd3";
        String final COLOR_RED = "#ffb6c1"
        String color = COLOR_BLUE;
        try {
            input = userInput.getText();
            response = duke.getResponse(input);
        } catch (DukeException e) {
            color = COLOR_RED;
```
Should the colour codes be stored as descriptively named final attributes for better understanding?

Reply:
Thats a good idea! Thanks 😄